### PR TITLE
Add Drupal-related VSCode extensions and settings.

### PR DESCRIPTION
**Pull request by paul121**
# The Problem/Issue/Bug

Addresses #6.

## How this PR Solves The Problem

Adds 3 of the extensions and default settings listed at https://www.drupal.org/docs/develop/development-tools/configuring-visual-studio-code

These docs recommend other extensions but it seems that these are not available on the Open VSX registry: https://open-vsx.org/

Notable extensions that are missing are:
- phpcs
- phpcbf
- composer

We might need to request the extension authors that these get added, or it looks like there may be a method of adding these to the registry too. More info here:https://www.gitpod.io/docs/vscode-extensions#where-do-i-find-extensions

## Related Issue Link(s)

See #6

### CODE REVIEW

**Comment by paul121** [comment]
Oops, it looks like the first commit had some other unintentional changes. Must admit I am pretty new to using the VSCode Git UI, so I think that just slipped in without me noticing! Will try and clean that up later.

On the bright side, I made this PR entirely from a GitPod :partying_face:

**Comment by paul121** [comment]
> Oops, it looks like the first commit had some other unintentional changes.

I force pushed to clean up those changes so I think this is ready for review. This could probably be expanded on further but I think this is a good start!

**Comment by shaal** [comment]
@paul121 Thank you so much!

It seems that there are a lot of [recommended vscode extensions](https://www.drupal.org/docs/develop/development-tools/configuring-visual-studio-code#s-recommended-extensions) that are still missing.

@paul121 What do you think about following the instructions here -
https://github.com/open-vsx/publish-extensions
To publish the missing vscode extension in Open VSX, and then they should be available there.

**Comment by paul121** [comment]
Cool, I was wondering about that. I'll try and start that soon

**Comment by shaal** [comment]
Thank you so much!

**Comment by paul121** [comment]
Just some notes on status of these extensions...

TLDR; I'm just going to focus on phpcs, phpcbf and composer like I mentioned in my first comment. Everything else seems to be outdated at this point.

### General
- phpcs
  - Drupal.org links to this one, but has been unmaintained for 3 years: https://github.com/ikappas/vscode-phpcs
  - Two alternatives already on Open VSX: https://open-vsx.org/?search=phpcs&category=&sortBy=relevance&sortOrder=desc
    - Maybe our best bet, most active: https://open-vsx.org/extension/wongjn/php-sniffer (supports both phpcs and phpcbf!)
  - Potential alternative that seems more active (but not on Open VSX): https://github.com/valeryan/vscode-phpsab
  
- phpcbf
  - Drupal.org links to this one, no activity for 2 years: https://github.com/soderlind/vscode-phpcbf
  - Already on Open VSX: https://open-vsx.org/extension/wongjn/php-sniffer (supports both phpcs and phpcbf!)

- Debugger for chrome (deprecated) - drupal.org links here: https://marketplace.visualstudio.com/items?itemName=msjsdiag.debugger-for-chrome

  > This extension has been deprecated as Visual Studio Code now has a bundled JavaScript Debugger that covers the same functionality. 

- PHP DocBlocker (included with this PR)
- Empty indent
  - DO links here, but no activity in 5 years: https://github.com/DmitryDorofeev/vscode-empty-indent
  - I feel like this feature would be covered in vscode core, phpcs, or other?
- PHP Debugger (already added)

### Drupal 8
- Composer
  - Drupal.org links to this one: https://github.com/ikappas/vscode-composer
    -  Existing issue to publish on Open VSX: https://github.com/ikappas/vscode-composer/issues/15
- Twig language 2 (included with this PR)
- Drual 8 snippets (drupal, javascript, and twig) all inactive for 3+ years

### Drupal 7
- Lets not go there

**Comment by paul121** [comment]
I think the `php-sniffer` extension will work great for both phpcs and phpcbf integration in vscode!: https://open-vsx.org/extension/wongjn/php-sniffer.

I installed and configured drupal/coder according to these docs: https://www.drupal.org/docs/contributed-modules/code-review-module/installing-coder-sniffer

Now I'm curious if we she could configure an alias for running `phpcs` as recommended here: https://www.drupal.org/docs/contributed-modules/code-review-module/php-codesniffer-command-line-usage#alias

**Comment by paul121** [comment]
I opened a PR to add the `vscode-composer` extension to open-vsx: https://github.com/open-vsx/publish-extensions/pull/410

The latest release was in 2017.

It seems like this may be another alternative (not on open-vsx) but I have not tested: https://github.com/faelv/composer-companion
